apps/prairielearn/src/ee/routers/lti13.ts

```diff
@@ -29,9 +42,10 @@ router.use(
   }),
 );
 
-router.use('/:lti13_instance_id/auth', lti13Auth);
+router.use('/:lti13_instance_id/auth', SSORequired, lti13Auth);
```

Comment:
I can't comment on the line, but: `lti13InstancePages` and the "catch all for some small pages" made me have to go and check what was actually handled by that router. Maybe name that `lti13InstanceMetadata` (borrowing the notion of a "metadata" URL from SAML) or something like that?

You might also consider reorganizing things so that you mount the non-authed config/jwks pages first, then mount "SSORequired", which would apply to all other pages. You can leave a comment too. We use this pattern in some places in `server.js` IIRC.

Reply:
c2468600b changed how these functions work - eliminating the `lti13InstancePages` generic route and adding specific routes for `jwks` and `config` above a middleware that checks for SSO. That seemed much cleaner.

I wasn't sure where to hang the functions that supply those `jwks` and `config` routes. Should they be individual pages? That seems like a lot of overhead for two small functions. It also didn't feel right to put them in `routers/lti13.ts`. Lib seemed logical but I didn't want to put more into `lib/lti13.ts` so I made a `lib/lti13Metadata.ts` file. Is that reasonable?

Reply:
We definitely don't want route handlers in the `lib` directory. Yes, I think these should be pages (`lti13Config` and `lti13Jwks`).